src/guide/data-methods.md

```diff
@@ -0,0 +1,137 @@
+
+```js
+const app = Vue.createApp({
+  data() {
+    return { count: 4 }
+  }
+})
+
+const vm = app.mount('#app')
+
+console.log(vm.$data.count) // => 4
+console.log(vm.count)       // => 4
+
+// Assigning a value to vm.count will also update $data.count
+vm.count = 5
+console.log(vm.$data.count) // => 5
+
+// ... and vice-versa
+vm.$data.count = 6
+console.log(vm.count) // => 6
+```
+
+These instance properties are only added when the instance is first created, so you need to ensure they are all present in the object returned by the `data` function. Where necessary, use `null`, `undefined` or some other placeholder value for properties where the desired value isn't yet available.
+
+It is possible to add a new property directly to the component instance without including it in `data`. However, because this property isn't backed by the reactive `$data` object it won't automatically be tracked by [Vue's reactivity system](reactivity.html).
+
+Vue uses a `$` prefix when exposing its own built-in APIs via the component instance, e.g. `$emit`. It also reserves the prefix `_` for internal properties. You should avoid using names for top-level `data` properties that start with either of these characters.
+
+## Methods
+
+You might be tempted to put functions in your `data` so that you can use them as methods:
+
+```js
+const app = Vue.createApp({
+  data() {
+    return {
+      count: 4,
+
+      // Don't do this
+      increment() {
+        this.count++
+      }
+    }
+  }
+})
+
+const vm = app.mount('#app')
+
+console.log(vm.count) // => 4
+
+vm.increment()
+
+console.log(vm.count) // => 5
+```
+
+This does work but there are a few problems with it:
+
+1. Methods typically aren't reassigned after creation, so there's no need for Vue's reactivity system to track them.
+2. If a function becomes detached from its object it will lose its `this` binding. This frequently happens when methods are used as event listeners.
+3. For components with several methods they will quickly clutter up the `data` function.
+
+To solve these problems, Vue has a separate configuration option especially for `methods`:
+
+```js
+const app = Vue.createApp({
+  data() {
+    return { count: 4 }
+  },
+  methods: {
+    increment() {
+      this.count++
+    }
+  }
+})
+
+const vm = app.mount('#app')
+
+console.log(vm.count) // => 4
+
+// Detach the function from `vm`
+const inc = vm.increment
+
+// Vue automatically binds the functions in `methods`,
+// so the `this` value inside the function will still
+// refer to the correct component instance
+inc()
+
+console.log(vm.count) // => 5
+```
+
+Just like all other properties of the component instance, the `methods` are accessible from within the component's template. If a method is called by a component's template, any reactive data it uses will be tracked and added as a rendering dependency.
+
+### Debouncing and Throttling
```

Comment:
**suggestion**: I would recommend moving this section to `Examples`. We try to reduce the cognitive load (especially on **Essentials** part and here we
1. Introduce a new concept (debouncing)
2. Introduce a new library (lodash)
3. Neither of these is directly connected to Vue

Reply:
Yeah, I had expected this to be controversial and tried to explain my reasoning when I first created the PR. Shame I failed to make the comments visible but the relevant one is here:

https://github.com/vuejs/docs-next/pull/514/files/952f8cba80365431108960ee9c957dca55a36f47#r491641224

In addition to what I wrote there:

1. Having it in the examples section wouldn't achieve everything I was trying to achieve. (See my earlier comment)
2. There is already something in the examples section that uses lodash for debouncing. https://v3.vuejs.org/examples/markdown.html. Unfortunately, because it's such a simple example it uses the 'cheap and cheerful' approach to debouncing that won't actually work with reusable components. Constructing a full example that 'does it properly' is actually quite tricky but that current example does somewhat lead people astray without anything to warn them.
3. The Vue 2 docs did have another debouncing example: https://vuejs.org/v2/guide/computed.html#Watchers. It seems that's been tweaked to remove the debouncing aspect in Vue 3.
4. I suspect that for most intermediate JS developers, debouncing would not be a new concept. One reason I also included the term 'throttling' was to jog the memories of those who had forgotten what debouncing was. For those who still don't know what debouncing is, I tried to word the opening sentence in such a way that it implied it could be skipped.

Specifically on the topic of debouncing itself, it is something that gets asked about in the context of Vue on a very regular basis. Just today someone was asking about it on SO: https://stackoverflow.com/questions/64026366. A quick search for questions mentioning Vue and debouncing throws up loads of other examples. One of the most popular Vue questions is https://stackoverflow.com/questions/42199956, which does have all the information lurking in the answers but you really have to dig through a lot of weak suggestions to find the gold.

All my waffling aside, the TLDR is that I'd like to keep that example where I currently have it. If you really want it gone then I will, of course, remove it.